packages/xchain-ethereum/src/etherscan-api.ts

```diff
@@ -17,3 +27,156 @@ export const getGasOracle = (baseUrl: string, apiKey?: string): Promise<GasOracl
 
   return axios.get(url + getApiKeyQueryParameter(apiKey)).then((response) => response.data.result)
 }
+
+/**
+ * Get token balance
+ *
+ * @see https://etherscan.io/apis#tokens
+ *
+ * @param {string} baseUrl The etherscan node url.
+ * @param {string} address The address.
+ * @param {string} assetAddress The token contract address.
+ * @param {string} apiKey The etherscan API key. (optional)
+ * @returns {BigNumberish} The token balance
+ */
+export const getTokenBalance = (
+  baseUrl: string,
+  address: string,
+  assetAddress: string,
+  apiKey?: string,
+): Promise<BigNumberish> => {
+  const url = baseUrl + `/api?module=account&action=tokenbalance&contractaddress=${assetAddress}&address=${address}`
+
+  return axios.get(url + getApiKeyQueryParameter(apiKey)).then((response) => response.data.result)
+}
+
+/**
+ * Get ETH balance
```

Comment:
Just curious: Why do we need a custom `getETHBalance` function if [EtherscanProvider of ethers.js](https://github.com/ethers-io/ethers.js/blob/7f775f7ad634d40bbf2625cde142f7a61e241b8f/packages/providers/src.ts/etherscan-provider.ts) already [provide such method](https://github.com/ethers-io/ethers.js/blob/7f775f7ad634d40bbf2625cde142f7a61e241b8f/packages/providers/src.ts/etherscan-provider.ts#L237)? Same for history (via [getHistory](https://github.com/ethers-io/ethers.js/blob/7f775f7ad634d40bbf2625cde142f7a61e241b8f/packages/providers/src.ts/etherscan-provider.ts#L392)), tx details (via [getTransaction](https://github.com/ethers-io/ethers.js/blob/7f775f7ad634d40bbf2625cde142f7a61e241b8f/packages/providers/src.ts/etherscan-provider.ts#L284)) and maybe other functions...?